lib/kontena/machine/digital_ocean/cloudinit.yml

```diff
@@ -48,6 +52,12 @@ coreos:
         Name=weave*
 
         [Network]
+    - name: zz-default.network
+      runtime: false
+      content: |
+        # default should only match real network interfaces
+        [Match]
```

Comment:
Yes, you need to override the CoreOS `zz-default.network` to not match the weave interfaces. Given the DO OEM cloudconfi stuff, I assume overriding it with an empty network unit that never matches any interfaces would be enough, but it's probably safer to just limit the `[Match]` to the `ethX` interfaces.